src/hooks/useLazyStore.js

```diff
@@ -28,23 +28,37 @@ export default function useLazyStore(lazyProps, additionalData = {}) {
   const [state, setState] = useState(createInitialState)
   const stateRef = useRef(state)
 
-  useEffect(() => {
-    stateRef.current = state
-  }, [state])
+  const updateState = finalState => {
+    if (typeof finalState !== 'function') {
+      stateRef.current = finalState
+
+      return setState(finalState)
+    } else {
+      return setState(state => {
+        stateRef.current = finalState(state)
+
+        return finalState(state)
```

Comment:
Are you intentionally calling the function twice? Why not just `return stateRef.current`?

Reply:
Good catch!